unittests/unittest_radix_tree.cpp

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright (c) 2018 Contributors as noted in the AUTHORS file
+
+This file is part of 0MQ.
+
+0MQ is free software; you can redistribute it and/or modify it under
+the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+0MQ is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "../tests/testutil.hpp"
+
+#include <radix_tree.hpp>
+#include <stdint.hpp>
+
+#include <string>
+#include <unity.h>
+#include <unordered_set>
+#include <vector>
+
+void setUp ()
+{
+}
+void tearDown ()
+{
+}
+
+bool tree_add (zmq::radix_tree &tree, const std::string &key)
+{
+    unsigned char *data = new unsigned char[key.size ()];
```

Comment:
I know this is only test code but cant we just memcpy key.data() rather than doing  new/loop/delete

Reply:
So something like `memcpy(data, key.data(), key.size())` instead of the loop? I think that could work as well.

Reply:
not sure why you would want overhead of allocating memory just to delete in the same function.

Reply:
Oh I see the issue now, yes this would work if we pass the pointer as a const value. If we change the functions to use const and this should work; also we don't modify the data anyway so this looks like the right thing to do.

Reply:
Thanks, we're using `key.data()` directly now and the test functions are much cleaner.